konfetti/src/main/java/nl/dionsegijn/konfetti/models/Shape.kt

```diff
@@ -1,12 +1,97 @@
 package nl.dionsegijn.konfetti.models
 
-/**
- * Created by dionsegijn on 3/26/17.
- * Available shapes
- * RECT (rectangle)
- * Circle
- */
-enum class Shape {
-    RECT(),
-    CIRCLE()
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.RectF
+import android.graphics.drawable.Drawable
+
+interface Shape {
+    /**
+     * Draw a shape to `canvas`. Implementations are expected to draw within a square of size
+     * `size` and must vertically/horizontally center their asset if it does not have an equal width
+     * and height.
+     */
+    fun draw(canvas: Canvas, paint: Paint, size: Float)
+
+    companion object {
+        // Maintain binary and backwards compatibility with previous enum API.
+        @JvmField
+        @Deprecated("Use Square class, instead.", replaceWith = ReplaceWith("Shape.Square"))
+        val RECT = Square
+
+        @JvmField
+        @Deprecated("Use Circle class, instead.", replaceWith = ReplaceWith("Shape.Circle"))
+        val CIRCLE = Circle
+    }
+
+    object Square : Shape {
+        override fun draw(
+            canvas: Canvas,
+            paint: Paint,
+            size: Float
+        ) {
+            canvas.drawRect(0f, 0f, size, size, paint)
+        }
+    }
+
+    class Rectangle(
+        /** The ratio of height to width. Must be within range [0, 1] */
+        private val heightRatio: Float
```

Comment:
I changed this and the other one in `DrawableShape` to be `heightRatio` instead of `sizeRatio` because it was confusing when I was trying to explain to a co-worker. Let me know if you prefer the old name or something else entirely.

Reply:
Sounds good to me 👍